src/utilities/color.ts

```diff
@@ -69,3 +69,19 @@ export function shadeColor(color: string, decimal: number): string {
 export function generateRandomPastelColor(): string {
   return hslToHex(360 * Math.random(), 25 + 70 * Math.random(), 82 + 10 * Math.random());
 }
+
+export function pickTextColorBasedOnBgColor(bgColor: string): 'dark' | 'light' {
+  const color = bgColor.charAt(0) === '#' ? bgColor.substring(1, 7) : bgColor;
+  const r = parseInt(color.substring(0, 2), 16); // hexToR
+  const g = parseInt(color.substring(2, 4), 16); // hexToG
+  const b = parseInt(color.substring(4, 6), 16); // hexToB
+  const uicolors = [r / 255, g / 255, b / 255];
+  const c = uicolors.map(col => {
+    if (col <= 0.03928) {
```

Comment:
What is this mysterious magic number?

Reply:
*magic numbers haha

Reply:
Added a docstring with comments but this is from https://stackoverflow.com/a/41491220 and ultimately https://www.w3.org/WAI/WCAG21/Techniques/general/G18.html